quinn-proto/src/connection/spaces.rs

```diff
@@ -309,6 +310,13 @@ pub struct Retransmits {
     pub(super) retire_cids: Vec<u64>,
     pub(super) ack_frequency: bool,
     pub(super) handshake_done: bool,
+    /// Two notable things about `new_tokens`:
+    ///
+    /// - NEW_TOKEN frames from an old path are not retransmitted on a new path
+    /// - If a token is lost, a new randomly generated token is re-transmitted rather than the
+    ///   original. This is so that if both transmissions end up being received, the client won't
+    ///   risk sending the same token twice.
+    pub(super) new_tokens: Vec<SocketAddr>,
```

Comment:
> Come to think of it, this will only ever contain N copies of the same remote address, right?

Almost, but no. One has to keep in mind how to `impl BitOrAssign for Retransmits`. Conceivably, the `impl BitOrAssign for Retransmits` could be replaced with some other method, which also takes as an input the current path remote address, and thus allows the lhs and rhs to have their `new_tokens` filtered at that point to allow a maximum of one distinct `SocketAddr` to survive, in addition to doing the same filtering in `poll_transmit`. However, that would be a conceptual complication and spend more lines of code, in exchange for an optimization that probably is undetectable.

Reply:
Fair enough. Please document for posterity.